pennylane/queuing.py

```diff
@@ -11,8 +11,116 @@
+Note that even though :class:`~.QuantumTape` is a queuable object, it does not have a ``queue`` method.
+
+When an objects attempts to be queued, it sends itself to the :class:`~.QueuingManager`. The :class:`~.QueuingManager`
+is a global singleton class that facilitates placing objects in the queue. All of :class:`~.QueuingManager`'s methods
+and properties are class methods and properties, so all instances will access the same information.
+
+The :meth:`~.QueuingManager.active_context` is the queue where any new objects are placed.
+The :class:`~.QueuingManager` is said to be *recording* if an active context exists.
+
+Active contexts are :class:`~.AnnotatedQueue` instances. They are *context managers* where recording occurs
+with in a ``with`` block.
+
+Let's take a look at an example. If we query the :class:`~.QueuingManager` outside of an
+:class:`~.AnnotatedQueue`'s context, we can see that nothing is recording and no active context exists.
+
+>>> print("Are we recording? ", qml.QueuingManager.recording())
+Are we recording?  False
+>>> print("What's the active context?", qml.QueuingManager.active_context())
+What's the active context? None
+
+Inside of a context, we can see the active recording context:
+
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     print("Are we recording? ", qml.QueuingManager.recording())
+...     print("Is q the active queue? ",q is qml.QueuingManager.active_context())
+Are we recording?  True
+Is q the active queue?  True
+
+If we construct an operator inside the recording context, we can see is added to the queue.
+
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op = qml.PauliX(0)
+>>> q.queue
+[PauliX(wires=[0])]
+
+If an operator is constructed outside of the context, we can manually add it to the queue by
+calling the :meth:`~.Operator.queue` method:
+
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op.queue()
+>>> q.queue
+[PauliX(wires=[0])]
+
+An object can only exist up to *once* in the queue, so calling queue multiple times will
+not do anything.
+
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op.queue()
+...     op.queue()
+>>> q.queue
+[PauliX(wires=[0])]
+
+Instead, we need to queue a copy of the operator:
+
+>>> from copy import copy
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op.queue()
+...     copy(op).queue()
+>>> q.queue
+[PauliX(wires=[0]), PauliX(wires=[0])]
+
+This can instead be accomplished with the :func:`~.apply` function:
+
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     qml.apply(op)
+...     qml.apply(op)
+>>> q.queue
+[PauliX(wires=[0]), PauliX(wires=[0])]
+
+PennyLane has many nested operators, such as :class:`~.Adjoint`, :class:`~.Pow`, :class:`~.Sum`,
+and others.  See the `qml.ops.op_math` module for more examples. Each operator in the composite will be queued.
```

Comment:
```suggestion
In the case of operators composed of other operators, each operator in the composite will be queued.
```
Or something like that. I'm ok to not make this change, but from my perspective: if they know what we're talking about, this is wordy. If they don't, they'll probably need to read some other docs before reading this anyway? Maybe I'm being too harsh...